typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -1294,6 +1295,56 @@
 
 [unsafe-undefined -Unsafe-Undefined]
 
+;; Section 4.21 (TreeLists)
+
+[treelist (-poly (a) (->* (list) a (-treelist a)))]
+[treelist-empty? (-poly (a) (-> (-treelist a) B))]
+[treelist-length (-poly (a) (-> (-treelist a) -Index))]
+[treelist-member?
+ (-poly (a)
+        (cl->* ((-treelist a) a . -> . Univ)
+               ((-treelist a) a (-> a a Univ) . -> . B)))]
+[treelist-first (-poly (a) (-> (-treelist a) a :T+ #f))]
```

Comment:
@bennn are the shallow annotations here right?

Reply:
yes 👍